pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -90,15 +90,12 @@ func (self *RefsHelper) CheckoutRef(ref string, options types.CheckoutRefOptions
 								if err := self.c.Git().Branch.Checkout(ref, cmdOptions); err != nil {
 									return err
 								}
-
-								onSuccess()
```

Comment:
One thing that perplexed me as I was copying the `CheckoutRef` code into the `NewBranch` function was this splitting of `onSuccess()` and `self.c.Refresh(refreshOptions)`. As best as I can tell, it makes no difference whether the pop operation is between them or before since the error return is always deferred to the end. The splitting was written this way since the code's inception: https://github.com/jesseduffield/lazygit/commit/e583cc251984deead48c062f19c3d88d88236eda#diff-50333022d9da124874ac1fde9b40b8f8bc521ce745578e1da2ee88d150529384.

Copying the split-between logic into the `NewBranch` function made it more complex so I decided to keep them together. For code style consistency, I transferred back that structure into the existing `CheckoutRef` function.